apps/etterlatte-brev-api/src/main/kotlin/no/nav/etterlatte/brev/dokarkiv/DokarkivKlient.kt

```diff
@@ -74,39 +75,92 @@ class DokarkivKlient(private val client: HttpClient, private val url: String) {
     internal suspend fun oppdaterJournalpost(
         journalpostId: String,
         request: OppdaterJournalpostRequest,
-    ): OppdaterJournalpostResponse =
-        client.put("$url/$journalpostId") {
-            contentType(ContentType.Application.Json)
-            setBody(request)
-        }.body()
-
-    internal suspend fun feilregistrerSakstilknytning(journalpostId: String): String =
-        client.patch("$url/$journalpostId/feilregistrer/feilregistrerSakstilknytning") {
-            contentType(ContentType.Application.Json)
-        }.body()
-
-    internal suspend fun opphevFeilregistrertSakstilknytning(journalpostId: String): String =
-        client.patch("$url/$journalpostId/feilregistrer/opphevFeilregistrertSakstilknytning") {
-            contentType(ContentType.Application.Json)
-        }.body()
+    ): OppdaterJournalpostResponse {
+        val response =
+            client.put("$url/$journalpostId") {
+                contentType(ContentType.Application.Json)
+                setBody(request)
+            }
+
+        return if (response.status.isSuccess()) {
+            response.body()
+        } else {
+            logger.error("Feil oppsto på oppdater journalpost: ${response.bodyAsText()}")
+
+            throw ForespoerselException(
+                status = response.status.value,
+                code = "FEIL_VED_OPPDATERING_AV_JOURNALPOST",
+                detail = "En ukjent feil oppsto ved oppdatering av journalpost",
+            )
+        }
```

Comment:
```suggestion
        if (response.status.isSuccess()) {
            return response.body()
        }
        
        logger.error("Feil oppsto på oppdater journalpost: ${response.bodyAsText()}")
        throw ForespoerselException(
            status = response.status.value,
            code = "FEIL_VED_OPPDATERING_AV_JOURNALPOST",
            detail = "En ukjent feil oppsto ved oppdatering av journalpost",
        )
```

Vet ikke om denne flyter bedre? Kanskje bare meg som ikke helt liker hvordan return if ser ut alltid :p